File: src/model/Card.java
package model;

public class Card {
	private Suit suit;
	private Rank rank;

	public Card(Rank r, Suit s) {
		suit = s;
		rank = r;
	}

	@Override
	public String toString() {
		return "" + rank + suit;
	}

	public Rank getRank() {
		return rank;
	}

	public Suit getSuit() {
		return suit;
	}

	public enum Suit {
		SPADE(3),
		HEART(2),
		DIAMOND(1),
		CLUB(0);

		int num;

		private Suit(int i) {
			num = i;
		}

		public int toInt() {
			return num;
		}

		@Override
		public String toString() {
			switch(this) {
			case SPADE:
				return "\u2660";
			case HEART:
				return "\u2665";
			case DIAMOND:
				return "\u2666";
			case CLUB:
				return "\u2663";
			default:
				throw new RuntimeException("impossible situation");
			}
		}
	}

	public enum Rank implements Comparable<Rank> {
		TWO(2),
		THREE(3),
		FOUR(4),
		FIVE(5),
		SIX(6),
		SEVEN(7),
		EIGHT(8),
		NINE(9),
		TEN(10),
		JACK(11),
		QUEEN(12),
		KING(13),
		ACE(14);

		private Rank(int n) {
			number = n;
		}

		int number;

		@Override
		public String toString() {
			switch (this) {
			case ACE:
				return "A";
			case TWO:
			case THREE:
			case FOUR:
			case FIVE:
			case SIX:
			case SEVEN:
			case EIGHT:
			case NINE:
			case TEN:
				return Integer.toString(number);
			case JACK:
				return "J";
			case QUEEN:
				return "Q";
			case KING:
				return "K";
			default:
				throw new RuntimeException("impossible case");
			}
		}

		int toInt() {
			return number;
		}
	}

	public long toLong() {
		return ((long)rank.toInt() << 32) | (long)suit.toInt();
	}
}

File: PokerBot/Card.cpp

#include "Card.h"

Card::Card(){
    rank = 0;
    suit = 0;
}
Card::Card(int r, int s){
    rank = r;
    suit = s;
}
int Card::getRank(){
    return rank;
}

int Card::getSuit(){
    return suit;
}

Card::Card(char r, int s){
    switch ( r ) {
        case 'K':
            rank = 13;
            break;
        case 'Q':
            rank = 12;
            break;
        case 'J':
            rank = 11;
            break;
        case 'T':
            rank = 10;
            break;
        case 'A':
            rank = 14;
            break;
        default:
            rank = (int)r - 48;
            break;
    }
    suit = s;
}

string Card::toString(){
    string ret;
    switch ( rank ){
        case 13:
            ret += "K";
            break;
        case 12:
            ret += "Q";
            break;
        case 11:
            ret += "J";
            break;
        case 10:
            ret += "T";
            break;
        case 14:
            ret += "A";
            break;
        default:
            ret += (char)(rank + 48);
            break;
    }
    switch ( suit ){
        case 0:
            ret += "c";
            break;
        case 1:
            ret += "d";
            break;
        case 2:
            ret += "h";
            break;
        case 3:
            ret += "s";
            break;
        default:
            break;
    }
    return ret;
}


bool operator ==( Card c1, Card c2){
    return (c1.getRank() == c2.getRank() && c1.getSuit() == c2.getSuit());
}